### PR TITLE
Password encryption feature

### PR DESCRIPTION
**Pull request by loneil**
# Description

Generate the service client password on the API side and return it to the UI.
Use asymetric encryption to only reveal the generated PW to the user's current session.
Show password on a popup (and allow decryption) after successful config change.
https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-83

## Types of changes

New feature (non-breaking change which adds functionality)

Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have read the [CONTRIBUTING](CONTRIBUTING.md) doc
- [x] I have checked that unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation (if appropriate)

## Further comments

Documentation of password generation technique will be created in this issue
https://apps.nrs.gov.bc.ca/int/jira/browse/SHOWCASE-170

For RSA key creation the following library is used for now
https://github.com/wwwtyro/cryptico

Password generation is done with the following library
https://www.npmjs.com/package/generate-password

### CODE REVIEW

File: backend/components/appConfig.js [hunk 1]
```diff
@@ -1,8 +1,47 @@
...
+  const path = '/applicationConfigurations';
+  const webAdeUrl = endpoint + path;
+  try {
+    const webAdeResponse = await axios.post(webAdeUrl, generatedConfig.webAdeCfg, {
+      headers: {
+        'Authorization': `Bearer ${token.access_token}`,
+        'Content-Type': 'application/json; charset=utf-8'
+      }
+    });
+    log.verbose(utils.prettyStringify(webAdeResponse.data));
```
Comment (jujaga) [inline backend/components/appConfig.js]: `log.verbose(arguments.callee.name, utils.prettyStringify(webAdeResponse.data));`

File: backend/components/utils.js [hunk 2]
```diff
@@ -26,6 +28,129 @@ const utils = {
...
+        newAppCfg.serviceClients[0].authorizations = [];
+      } else {
+        newAppCfg.actions = [
+          {
+            name: `${newAppCfg.applicationAcronym}_ACTION`,
+            description: `${newAppCfg.applicationAcronym} action`,
+            privilegedInd: false
+          }
+        ];
+        newAppCfg.roles = [
```
Comment (jujaga) [inline backend/components/utils.js]: add newline above

File: frontend/src/components/ConfigForm.vue [hunk 3]
```diff
@@ -135,8 +117,50 @@
...
+            </v-card-actions>
+          </v-card>
+        </v-dialog>
+
+        <v-dialog v-model="passwordDialog" persistent max-width="600">
+          <v-card>
+            <v-card-title class="headline">Application Configuration Updated</v-card-title>
+            <v-card-text>
+              <p>
+                You application configuration for
```
Comment (jujaga) [inline frontend/src/components/ConfigForm.vue]: Your

File: frontend/src/components/ConfigForm.vue [hunk 4]
```diff
@@ -191,60 +217,65 @@ export default {
...
 
       // this is temporary, only allow MSSC to be used at the moment
-      if (this.userAppCfg.applicationAcronym !== 'MSSC') {
+      if (
+        this.userAppCfg.applicationAcronym !== 'MSSC' &&
+        this.userAppCfg.applicationAcronym !== 'DOMO'
+      ) {
         this.displayMessage(
           false,
           'Temp: Only the application acronym MSSC is supported for now.'
```
Comment (jujaga) [inline frontend/src/components/ConfigForm.vue]: MSSC and DOMO

File: backend/components/appConfig.js (hunk 1 above)
Comment (loneil) [reply backend/components/appConfig.js]: Fixed

File: backend/components/utils.js (hunk 2 above)
Comment (loneil) [reply backend/components/utils.js]: Fixed

File: frontend/src/components/ConfigForm.vue (hunk 3 above)
Comment (loneil) [reply frontend/src/components/ConfigForm.vue]: Fixed

File: frontend/src/components/ConfigForm.vue (hunk 4 above)
Comment (loneil) [reply frontend/src/components/ConfigForm.vue]: Fixed

**Pull request merged**